app/src/main/java/org/p2p/wallet/auth/interactor/restore/CustomShareRestoreInteractor.kt

```diff
@@ -27,15 +27,17 @@ class CustomShareRestoreInteractor(
         val temporaryUserPrivateKey = restoreFlowDataLocalRepository.userRestorePrivateKey
             ?: throw RestoreWalletFailure("User restore private key is null")
 
-        if (isResend || userPhoneNumber != restoreFlowDataLocalRepository.userPhoneNumber) {
+        val isNumberEquals = userPhoneNumber == restoreFlowDataLocalRepository.userPhoneNumber
+        val isRestoreWalletRequestSent = restoreFlowDataLocalRepository.isRestoreWalletRequestSent
+        if (isResend || (!isRestoreWalletRequestSent && !isNumberEquals)) {
```

Comment:
condition (isResend || (!isRestoreWalletRequestSent && !isNumberEquals)) -> to a variable called `isRestoreRequestNeeded`